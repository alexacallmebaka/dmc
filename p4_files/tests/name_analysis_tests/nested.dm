f : (arg1 : int,arg2 : bool) int {
}

g : () void {
	a : int;
  if (true) {
    b : int;
  } else {
    b : int = 1;
    give b;
  }
	f(a, true);
	g();
}
