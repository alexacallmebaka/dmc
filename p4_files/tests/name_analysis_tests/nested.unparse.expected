f{(int, bool)->int} : (arg1{int} : int, arg2{bool} : bool) int  {
}
g{()->void} : () void  {
    a{int} : int;
    if (true){
        b{int} : int;
    } else {
        b{int} : int = 1;
        give b{int};
    }
    f{(int, bool)->int}(a{int}, true);
    g{()->void}();
}
